### PR TITLE
NIAD-151: Adding duplicate key to throw specific exception

### PR DESCRIPTION
**Pull request by bartek-sarul**

### CODE REVIEW

File: integration-tests/integration_tests/Pipfile [hunk 1]
```diff
@@ -6,10 +6,11 @@ verify_ssl = true
 [dev-packages]
 coverage = "~=4.5"
 unittest-xml-reporting = "~=2.5"
+#integration-adaptors-common = {editable = true, path = "../../../integration-adaptor-common"}
```
Comment (grzegorz-robak) [inline integration-tests/integration_tests/Pipfile:9]: this is needed? (applies to all pipfiles)

Comment (bartek-sarul) [reply integration-tests/integration_tests/Pipfile:9]: Yes, I've explicitly left it here. This is a temporary workaround for multi-pipfile
User can uncomment that and install into local venv
It helps a lot with development on commons

Comment (grzegorz-robak) [reply integration-tests/integration_tests/Pipfile:9]: can you put that info in there?

Comment (bartek-sarul) [reply integration-tests/integration_tests/Pipfile:9]: I'll do that in the other PR where I'll create a single Pipfile

Comment (bartek-sarul) [reply integration-tests/integration_tests/Pipfile:9]: This approach will still be valid for a single pipfile

**Pull request merged**